### ISSUE TITLE
Crash from a combination of newlines and unicode characters in the statusline

### ISSUE THREAD

**Issue by nyz93** [neovim/neovim#17969]
### Neovim version (nvim -v)

NVIM v0.7.0-dev+1383-g2a4660014

### Vim (not Nvim) behaves the same?

no, vim 8.2.4464-1

### Operating system/version

ArchLinux

### Terminal name/version

alacritty 0.10.0-dev (3371851c)

### $TERM environment variable

screen-256color

### Installation

build from repo

### How to reproduce the issue

Using the following configuration:
```vim
function Crash()
    return "\n\n%##"
endfunction

set statusline=%!Crash()
```

### Expected behavior

Neovim shows the correct statusline.

The mentioned vim version does not crash but instead shows a slightly corrupted statusline: `^@^@  ²`.

### Actual behavior

If the statusline contains at exactly two (using any more does not crash) newline characters, any number multibyte unicode characters, and a highlight group (in this specific order), then neovim crashes with a message: `nvim/ui_compositor.c:551: ui_comp_raw_line: Assertion 'attrs[i] >= 0' failed.`.
Interestingly a name for the highlight group is not required, crashes with or without one.

**Comment by zeertzjq** [comment]
Can you bisect?

**Comment by dundargoc** [comment]
Surely the expected behavior should be that is show the correct statusline? :) The assertions are only triggered on a debug build. If you build neovim with `CMAKE_BUILD_TYPE=Release` you'll see a similarly corrupted statusline, but neovim won't crash.

**Comment by zeertzjq** [comment]
The crash to not specific to newline, it applies to any non-printable character.

**Comment by zeertzjq** [comment]
If `` is changed to `▎`, it crashes with one newline as well.

**Comment by zeertzjq** [comment]
https://github.com/vim/vim/issues/11772

**Comment by zeertzjq** [comment]
Fixed by #11756
